Gemfile

```diff
@@ -11,43 +11,40 @@ gem 'jquery-rails'
 gem 'jquery-ui-rails'
 gem 'mysql2', '~> 0.3.18'
 gem 'nested_form'
+gem 'paperclip'
 gem 'popper_js', '~> 1.14', '>= 1.14.5'
 gem 'rails', '4.2.0'
 gem 'rails-jquery-tokeninput'
 gem 'rails4-autocomplete'
-gem 'rspec-rails', '~> 3.8'
 gem 'sass', '~> 3.7', '>= 3.7.4'
 gem 'sass-rails', '~> 5.0'
 gem 'sassc-rails', '>= 2.1.0'
 gem 'sdoc', '~> 0.4.0', group: :doc
 gem 'select2-rails'
 gem 'sequenceid'
 gem 'uglifier', '>= 1.3.0'
+gem 'will_paginate', '~> 3.1.0'
+gem 'will_paginate-bootstrap4'
 gem 'wicked_pdf', '~>1.4.0'
 gem 'wkhtmltopdf-binary', '~>0.12.4'
 
-group :development do
+group :development, :test do
   gem 'better_errors'
   gem 'capybara', '~> 2.5'
+  gem 'clockwork'
   gem 'database_cleaner'
+  gem 'delayed_job_active_record'
   gem 'factory_girl_rails', '~> 4.5.0'
   gem 'pry', '=0.10.1'
   gem 'pry-rails', '=0.3.3'
+  gem 'rails-observers'
   gem 'rspec-rails', '~> 3.8'
   gem 'shoulda-matchers', require: false
   gem 'spring'
   gem 'web-console', '~> 2.0'
 end
 
 group :test do
-  gem 'better_errors'
```

Comment:
why are we removing all these gems?

Reply:
development and test common gems move in test development group... thats why..